src/SW_netCDF_Input.c

```diff
@@ -0,0 +1,8903 @@
+    }
+
+    // Geographic CRS attributes
+    fill_netCDF_with_geo_CRS_atts(
+        &SW_netCDFOut->crs_geogsc,
+        ncFileID,
+        SW_netCDFOut->coordinate_system,
+        geo_id,
+        LogInfo
+    );
+    if (LogInfo->stopRun) {
+        return; // Exit function prematurely due to error
+    }
+
+    // Projected CRS variable/attributes
+    if (!SW_netCDFOut->primary_crs_is_geographic) {
+
+        /* Do not deflate crs_projsc */
+        SW_NC_create_netCDF_var(
+            &proj_id, "crs_projsc", NULL, ncFileID, NC_BYTE, 0, NULL, 0, LogInfo
+        );
+        if (LogInfo->stopRun) {
+            return; // Exit function prematurely due to error
+        }
+
+        fill_netCDF_with_proj_CRS_atts(
+            &SW_netCDFOut->crs_projsc, ncFileID, proj_id, LogInfo
+        );
+        if (LogInfo->stopRun) {
+            return; // Exit function prematurely due to error
+        }
+    }
+
+    // Write global attributes
+    fill_netCDF_with_global_atts(
+        SW_netCDFOut, ncFileID, domType, fx, isInputFile, LogInfo
+    );
+}
+
+/**
+@brief Fill the progress variable in the progress netCDF with values
+
+@param[in] SW_Domain Struct of type SW_DOMAIN holding constant
+    temporal/spatial information for a set of simulation runs
+@param[in,out] LogInfo Holds information on warnings and errors
+*/
+static void fill_prog_netCDF_vals(SW_DOMAIN *SW_Domain, LOG_INFO *LogInfo) {
+
+    int domVarID = SW_Domain->netCDFInput.ncDomVarIDs[vNCdom];
+    int progVarID = SW_Domain->netCDFInput.ncDomVarIDs[vNCprog];
+    unsigned int domStatus;
+    unsigned long suid;
+    unsigned long ncSuid[2];
+    unsigned long nSUIDs = SW_Domain->nSUIDs;
+    unsigned long nDimY = SW_Domain->nDimY;
+    unsigned long nDimX = SW_Domain->nDimX;
+    int progFileID = SW_Domain->SW_PathInputs.ncDomFileIDs[vNCprog];
+    int domFileID = SW_Domain->SW_PathInputs.ncDomFileIDs[vNCdom];
+    size_t start1D[] = {0};
+    size_t start2D[] = {0, 0};
+    size_t count1D[] = {nSUIDs};
+    size_t count2D[] = {nDimY, nDimX};
+    size_t *start =
+        (strcmp(SW_Domain->DomainType, "s") == 0) ? start1D : start2D;
+    size_t *count =
+        (strcmp(SW_Domain->DomainType, "s") == 0) ? count1D : count2D;
+
+    signed char *vals = (signed char *) Mem_Malloc(
+        nSUIDs * sizeof(signed char), "fill_prog_netCDF_vals()", LogInfo
+    );
+    if (LogInfo->stopRun) {
+        return; // Exit function prematurely due to error
+    }
+
+    for (suid = 0; suid < nSUIDs; suid++) {
+        SW_DOM_calc_ncSuid(SW_Domain, suid, ncSuid);
+
+        SW_NC_get_single_val(
+            domFileID,
+            &domVarID,
+            NULL,
+            ncSuid,
+            (void *) &domStatus,
+            "unsigned integer",
+            LogInfo
+        );
+        if (LogInfo->stopRun) {
+            goto freeMem; // Exit function prematurely due to error
+        }
+
+        vals[suid] = (domStatus == NC_FILL_UINT) ? NC_FILL_BYTE : PRGRSS_READY;
```

Comment:
fill_prog_netCDF_vals(): the check `(domStatus == NC_FILL_UINT)` does not work correctly for a "domain.nc" that was produced (e.g., outside SOILWAT2) with a "domain" variable that is not unsigned int.

--> can we query the type of "domain" variable for the call SW_NC_get_single_val() and obtain the _FillValue attribute of the "domain" variable (if available) to check  `(domStatus == fillValue)`